BUILD.api

```diff
@@ -118,3 +118,55 @@ genrule(
     message = "Applying overrides to cfg proto",
     tools = ["@org_golang_x_tools_imports//:goimports"],
 )
+
+filegroup(
+    name = "mixer/v1/attributes_file",
+    srcs = ["mixer/v1/global_dictionary.yaml"],
+    visibility = ["//visibility:private"],
+)
+
+genrule(
+    name = "mixer/v1/attributes_list",
+    srcs = [":mixer/v1/attributes_file"],
+    outs = ["attributes_list.txt"],
+    cmd = "grep -e '^- ' $(location :mixer/v1/attributes_file) | sed -e 's/- /\"/g' -e 's/$$/\",/g' > $@",
+    message = "Extracting strings",
+    visibility = ["//visibility:private"],
+)
+
+PRE = """package attributes; func GlobalList() ([]string) { tmp := make([]string, len(globalList)); copy(tmp, globalList); return tmp;}; var ( globalList = []string{"""
```

Comment:
Would it be possible to at least write the code in proper python or shell, and just call that from bzl ? 
It is easier to maintain, test, read, debug, understand.

Bazel is a great build tool - but not a programming language :-)